bake/hclparser/hclparser.go

```diff
@@ -25,11 +25,17 @@ type Opt struct {
 }
 
 type variable struct {
-	Name        string         `json:"-" hcl:"name,label"`
-	Default     *hcl.Attribute `json:"default,omitempty" hcl:"default,optional"`
-	Description string         `json:"description,omitempty" hcl:"description,optional"`
-	Body        hcl.Body       `json:"-" hcl:",body"`
-	Remain      hcl.Body       `json:"-" hcl:",remain"`
+	Name        string              `json:"-" hcl:"name,label"`
+	Default     *hcl.Attribute      `json:"default,omitempty" hcl:"default,optional"`
+	Description string              `json:"description,omitempty" hcl:"description,optional"`
+	Validation  *variableValidation `json:"validation,omitempty" hcl:"validation,block"`
```

Comment:
Shouldn't this be array type?

Reply:
Yes we should, implemented and added tests.